common/lib/xmodule/xmodule/modulestore/mongo/draft.py

```diff
@@ -830,6 +836,47 @@ def _verify_branch_setting(self, expected_branch_setting):
                 actual_setting=actual_branch_setting
             )
 
+    def _format_course_structure(self, course_key, structure):
+        course_block_key = unicode(BlockUsageLocator(course_key, 'course', course_key.run))
+        structure[course_block_key] = {
+            u'display_name': self.get_course(course_key).display_name,
+            u'children': []
+        }
+
+        for key, block in structure.iteritems():
+            # Set block_type attribute using the locator key
+            block_locator = BlockUsageLocator.from_string(key)
+            block[u'block_type'] = block_locator.block_type
+
+            # Ensure all required fields have values
+            block[u'display_name'] = block.get(u'display_name', None)
+            block[u'format'] = block.get(u'format', None)
+            block[u'graded'] = block.get(u'graded', False)
+
+            # Create a link from the parent to this block and remove this block's link to the parent.
+            parent = block.pop(u'parent', {})
+            parent_key = parent.get(ModuleStoreEnum.Branch.published_only) \
+                         or parent.get(ModuleStoreEnum.Branch.draft_preferred)
+            if parent_key:
+                children = structure[parent_key].get(u'children', [])
```

Comment:
Nit: if you use `structure[parent_key].setdefault(u'children', [])`, then you can `append` without having to reassign the list.